src/cpp/src/logger.hpp

```diff
@@ -2,16 +2,89 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #pragma once
-#include <iostream>
+#include <atomic>
+#include <cstdarg>
+#include <memory>
+#include <mutex>
+#include <ostream>
 #include <string>
+#include <string_view>
+#include <type_traits>
+#include <utility>
+
+#include "openvino/openvino.hpp"
 
 namespace ov::genai {
+ov::log::Level get_openvino_env_log_level();
+ov::log::Level get_cur_log_level();
 
 class Logger {
 public:
-    static void warn(const std::string& message) {
-        std::cout << "[WARN] " << message << '\n';
-    };
+    Logger();
```

Comment:
There's already `LogHelper` in `openvino/src/common/util/include/openvino/util/log.hpp`. Would it make sense to reuse it?